src/markdown-pages/collect-data/opentelemetry-manual/instrument.mdx

```diff
@@ -0,0 +1,641 @@
+    except KeyError as ke:
+        msg = f"Key `{key}` doesn't exist"
+        logging.debug(msg)
+        raise KeyDoesNotExistError(msg)
+
+def create(key, value):
+    """Write key:value to the database."""
+    global db
+
+    if key in db:
+        msg = f"Key `{key}` already exists"
+        logging.debug(msg)
+        raise DuplicateKeyError(msg)
+
+    db[key] = value
+    logging.debug("Successful create")
+    return value
+
+def update(key, value):
+    """Update key in the database."""
+    global db
+
+    if key in db:
+        db[key] = value
+        logging.debug("Successful update")
+        return value
+
+    msg = f"Key `{key}` doesn't exist"
+    logging.debug(msg)
+    raise KeyDoesNotExistError(msg)
+
+def delete(key):
+    """Delete key from the database."""
+    global db
+
+    if key in db:
+        del db[key]
+        logging.debug("Successful delete")
+        return True
+
+    return False
+```
+
+</>
+
+<br />
+
+A span processor processes span data before exporting it to a telemetry consumer. The `BatchSpanProcessor` you used here batches spans before exporting them. This reduces the number of requests you send to New Relic.
+
+Within this span processor, you also configured a [span exporter](https://opentelemetry.io/docs/reference/specification/trace/sdk/#span-exporter). The exporter is in charge of serializing and sending spans to the consumer. Here, you used [OTLP](https://opentelemetry.io/docs/reference/specification/protocol/), OpenTelemetry's exchange protocol, and [Gzip](https://www.gnu.org/software/gzip/) compression to efficiently transport your telemetry data to New Relic.
+
+You configure your tracer provider with this processor logically prior to setting your tracer provider in the API.
+
+</Step>
+
+<Step>
+
+Create two environment variables that you use to configure your OpenTelemetry pipelines. Don't forget to replace the license key placeholder with your [real one](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#ingest-license-key):
+
+<>
+
+```bash
+export OTEL_EXPORTER_OTLP_ENDPOINT=https://otlp.nr-data.net:4317
+export OTEL_EXPORTER_OTLP_HEADERS="api-key=<YOUR-LICENSE-KEY>"
+```
+
+The [OpenTelemetry Protocol](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md) (OTLP) endpoint is the url of our OpenTelemetry receiver. Your service sends data directly to New Relic through this endpoint. https://otlp.nr-data.net:4317 is our [US endpoint](https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-quick-start/#review-settings). If you're in the EU, use https://otlp.eu01.nr-data.net/ instead.
+
+There are several different types of API keys to choose from in New Relic that each serve a different purpose. To instrument your application with OpenTelemetry, you need a [license key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#ingest-license-key).
```

Comment:
There's already reference to licence key. Do we need a second reference? If so, could it be moved to callout?